src/repos/UserRepo.ts

```diff
@@ -0,0 +1,72 @@
+import { v4 as uuidv4 } from 'uuid';
+import { User, UserModel } from '../entities/User';
+import { PublicationID } from '../common/types';
+
+/**
+ * Create new pseudo suser.
+ */
+const createUser = async (
+  deviceToken: string,
+  followedPublicationsIDs: string[],
+  deviceType: string,
+): Promise<User> => {
+  // create PublicationID obejcts from string of followed publications
+  const followedPublications = followedPublicationsIDs.map((id) => {
+    return Object.assign(new PublicationID(), { id });
+  });
+
+  const uuid = uuidv4();
+  const newUser = Object.assign(new User(), {
+    uuid,
+    deviceToken,
+    followedPublications,
+    deviceType,
+  });
+
+  return UserModel.create(newUser);
+};
+
+/**
+ * Adds publication of pubID to user represented by uuid's followedPublications.
+ */
+const followPublication = async (uuid: string, pubID: string): Promise<User> => {
+  const user = await UserModel.findOne({ uuid });
+  user.followedPublications.push(Object.assign(new PublicationID(), { id: pubID }));
```

Comment:
I think it's worth checking if the pubID is valid before creating it?

Reply:
I think it's okay not to check if the pubID is valid for the following reasons: 1. PubIDs are only ever made on Publication object creation which happens once during bootstrap 2. All queries that we have that use pubIDs to query articles just return an empty array if the pubID is invalid so there already exists a check to avoid consequences from invalid pubIDs. I don't think it's a bad idea to add an extra layer of protection against invalid pubIDs, but I don't know how to check validity without some how querying the DB or scanning the publication json for a list of pubIDs and I'd like to avoid that expense if possible.